tracing/src/main/java/com/palantir/tracing/Tracer.java

```diff
@@ -100,6 +100,9 @@ public static void initTrace(Optional<Boolean> isObservable, String traceId) {
 
     /**
      * Initializes the current thread's trace, erasing any previously accrued open spans.
+     *
+     * The new trace will be persistent for the duration of the next span opened on it via
+     * {@link #startSpan}, and will be discarded when that span is completed or discarded.
```

Comment:
Instead of documenting effects, perhaps we should document how this is meant to be used correctly? This is only supposed to be called just before starting a new span, and only to set trace state when a request is received.

Reply:
Yeah, there was some slack discussion on this being a fairly internal API. Happy to see other fixes, including possibly just changing this method to take a span name and initialise the span immediately, this was just a quick fix for something that bit me.